pkg/identity/cache/allocator.go

```diff
@@ -204,6 +222,35 @@ func AllocateIdentity(ctx context.Context, lbls labels.Labels) (*identity.Identi
 		return nil, false, err
 	}
 
+	identityRefCountMutex.Lock()
+	refCountNew := idPoolRefCount[id] == 0
+	if refCountNew {
+		identityControllerManager.UpdateController(fmt.Sprintf("sync-identity (%d)", id),
+			controller.ControllerParams{
+				DoFunc: func(ctx context.Context) error {
```

Comment:
Should this have a `StopFunc` to release the identity? I'm not sure if the following is possible here:

* `Release` is called, which releases the identity
* Since the controller's `DoFunc` is ran as a goroutine, this controller could then runs and allocate the identity again
* The controller is then removed

In this case, would the identity still be allocated?

Reply:
Yes it will, perhaps the `Release` identity should be performed in 2 steps, one to decrease the number of ref count and remove the controller before actually releasing the identity. If the `identity` release fails we re-add the controller back again.